apollo-backend/src/main/java/io/logz/apollo/services/SlaveService.java

```diff
@@ -0,0 +1,143 @@
+package io.logz.apollo.services;
+
+import com.google.common.base.Splitter;
+import io.logz.apollo.common.EnvironmentVariableGetter;
+import io.logz.apollo.configuration.ApolloConfiguration;
+import io.logz.apollo.dao.SlaveDao;
+import io.logz.apollo.models.Slave;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static io.logz.apollo.common.EnvironmentVariableGetter.getEnvVarOrProperty;
+import static java.util.Objects.requireNonNull;
+
+@Singleton
+public class SlaveService {
+    private static final Logger logger = LoggerFactory.getLogger(SlaveService.class);
+
+    public static final String SLAVE_PROPERTY = "slave";
+    public static final String SLAVE_CSV_ENVIRONMENTS = "environments";
+    private final String slaveId;
+    private final ScheduledExecutorService scheduledExecutorService;
+    private final Boolean isSlave;
+    private final List<Integer> environmentIds;
+    private final ApolloConfiguration apolloConfiguration;
+    private final SlaveDao slaveDao;
+
+    private boolean started = false;
+
+    @Inject
+    public SlaveService(ApolloConfiguration apolloConfiguration, SlaveDao slaveDao) {
+        this.slaveDao = requireNonNull(slaveDao);
+        this.apolloConfiguration = requireNonNull(apolloConfiguration);
+
+        isSlave = Boolean.parseBoolean(getEnvVarOrProperty(SLAVE_PROPERTY));
+        environmentIds = parseEnvironmentIds();
+
+        if (isSlave && environmentIds == null) {
```

Comment:
```suggestion
        if (isSlave && isBlank(environmentIds)) {
```

Reply:
which isBlank do you mean because environmentIds is List<Integer> and not a String?

Reply:
`CollectionUtils::isEmpty` from `org.apache.commons.collections4`

Reply:
@asafm